packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,222 @@
+# File Size Table Column
+
+## Overview
+
+The file size table column will provide a way to visualize numeric data that represents a file size (or other binary data size). It will automatically convert the raw value to use an appropriate display unit.
+
+### Background
+
+[Work Item](https://ni.visualstudio.com/DevCentral/_workitems/edit/2526114)
+
+### Non-goals
+
+-   Configurable number of decimal places
+-   Configurable units (e.g. display all values in KB)
+-   Grouping by value ranges/buckets
+
+### Features
+
+-   Converts the byte value to largest unit (e.g. KB, MB, GB) that results in a value of >=1
+-   Option to use 1024-based units (e.g. KiB) or 1000-based units (e.g. KB)
+-   Displays up to one decimal place
+-   Sorts/groups by numeric value (number of bytes) rather than display string
+-   Localized number and unit strings
+
+### Risks and Challenges
+
+### Prior Art/Examples
+
+Size column in [SLE Files grid](https://dev.lifecyclesolutions.ni.com/files/)
+
+![SLE Files grid](../spec-images/fileSizesSle.png)
+
+Windows explorer (details view)
+
+![Windows explorer](../spec-images/fileSizesWindows.png)
+
+Chrome Dev Tools Network tab
+
+![Chrome Dev Tools Network tab](../spec-images/fileSizeChromeDevTools.png)
+
+---
+
+## Design
+
+### API
+
+`nimble-table-column-file-size`
+
+-   `field-name` - name of the record field containing the data. Data must be a `number` byte count.
+-   `unit-type` - "binary" (KiB, MiB, GiB, TiB, PiB) or "decimal" (KB, MB, GB, TB, PB); defaults to "decimal"
```

Comment:
I'll point out that this is a difference in behavior from what exists in SLE -- particularly, file sizes there are displayed in `KB`, `MB`, `GB`, but they are determined using 1024 division into the original "bytes" value.

Referring back to a potential risk I mentioned in a different comment, I think we need to be careful about diverging too much from existing code since that code can be used in other places within an applications. Some thoughts about how to resolve this:
- Make sure POs are aligned with our behavior change (i.e. either starting to use different units or starting to show values converted to upper units using 1000 rather than 1024)
- Update the `format-bytes` pipe to match what the column SLE will use so that you don't see one value in the table and a different value when looking at the details of a file

Reply:
I will figure out which of these approaches to take when uptaking the new column in SLE.